src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/RuntimePlatformCheckAnalyzerTests.cs

```diff
@@ -0,0 +1,785 @@
+
+            if (RuntimeInformationHelper.IsOSPlatformOrLater(OSPlatform.Linux, 2, 0))
+            {
+                {|#2:M2()|};    // Platform checks:'IsOSPlatformOrLater;Linux;1 && IsOSPlatformOrLater;Linux;2.0'
+            }
+            else if (!RuntimeInformationHelper.IsOSPlatformEarlierThan(OSPlatform.Windows, 3, 1, 1))
+            {
+                {|#3:M2()|};    // Platform checks:'!IsOSPlatformEarlierThan;Windows;3.1.1 && !IsOSPlatformOrLater;Linux;2.0 && IsOSPlatformOrLater;Linux;1'
+            }
+
+            {|#4:M2()|};    // Platform checks:'IsOSPlatformOrLater;Linux;1'
+        }
+        else
+        {
+            {|#5:M2()|};    // Platform checks:'!IsOSPlatformOrLater;Linux;1'
+        }
+
+        {|#6:M2()|};        // Platform checks:''
+
+        if ({|#7:IsWindows3OrLater()|})    // Platform checks:''
+        {
+            {|#8:M2()|};    // Platform checks:''
+        }
+
+        {|#9:M2()|};        // Platform checks:''
+    }
+
+    void M2()
+    {
+    }
+
+    bool IsWindows3OrLater()
+    {
+        return RuntimeInformationHelper.IsOSPlatformOrLater(OSPlatform.Windows, 3, 0, 0, 0);
+    }
+}" + PlatformCheckApiSource;
+
+            await VerifyCS.VerifyAnalyzerAsync(source,
+                VerifyCS.Diagnostic(RuntimePlatformCheckAnalyzer.Rule).WithLocation(0).WithArguments(""),
+                VerifyCS.Diagnostic(RuntimePlatformCheckAnalyzer.Rule).WithLocation(1).WithArguments("IsOSPlatformOrLater;Linux;1"),
+                VerifyCS.Diagnostic(RuntimePlatformCheckAnalyzer.Rule).WithLocation(2).WithArguments("IsOSPlatformOrLater;Linux;1 && IsOSPlatformOrLater;Linux;2.0"),
+                VerifyCS.Diagnostic(RuntimePlatformCheckAnalyzer.Rule).WithLocation(3).WithArguments("!IsOSPlatformEarlierThan;Windows;3.1.1 && !IsOSPlatformOrLater;Linux;2.0 && IsOSPlatformOrLater;Linux;1"),
+                VerifyCS.Diagnostic(RuntimePlatformCheckAnalyzer.Rule).WithLocation(4).WithArguments("IsOSPlatformOrLater;Linux;1"),
+                VerifyCS.Diagnostic(RuntimePlatformCheckAnalyzer.Rule).WithLocation(5).WithArguments("!IsOSPlatformOrLater;Linux;1"),
+                VerifyCS.Diagnostic(RuntimePlatformCheckAnalyzer.Rule).WithLocation(6).WithArguments(""),
+                VerifyCS.Diagnostic(RuntimePlatformCheckAnalyzer.Rule).WithLocation(7).WithArguments(""),
+                VerifyCS.Diagnostic(RuntimePlatformCheckAnalyzer.Rule).WithLocation(8).WithArguments(""),
+                VerifyCS.Diagnostic(RuntimePlatformCheckAnalyzer.Rule).WithLocation(9).WithArguments(""));
+        }
+
+        [Fact]
+        public async Task DebugAssertAnalysisTest()
+        {
+            var source = @"
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+class Test
+{
+    void M1()
+    {
+        {|#0:M2()|};    // Platform checks:''
+
+        {|#1:Debug.Assert(RuntimeInformationHelper.IsOSPlatformOrLater(OSPlatform.Windows, 3, 0, 0, 0))|};  // Platform checks:'IsOSPlatformOrLater;Windows;3.0.0.0'
```

Comment:
@mavasani `Debug.Assert` is good one that I didn't consider. For control flow, are you special casing `Debug.Assert` or are you handling all methods marked with `[DoesNotReturn]`, `[DoesNotReturnIf]`? What about `Environment.FailFast`, `Environment.Exit` etc?
//cc @buyaa-n

Reply:
Right now I just handled Assert methods in Debug and Contracts namespace (latter was already handled for a different analyzer). It would be trivial to enhance this with functionality to handle more cases - @buyaa-n probably good to make a note of these cases in the TODO list.

Reply:
Yes, there is many test cases i didn't think about, it even supports InterproceduralAnalysis

Reply:
> Right now I just handled Assert methods in Debug and Contracts namespace (latter was already handled for a different analyzer). It would be trivial to enhance this with functionality to handle more cases - @buyaa-n probably good to make a note of these cases in the TODO list.

Sure i will add it in my TODO list

Reply:
> it even supports InterproceduralAnalysis

Yes, that comes for free for any analysis based on the repo's framework. However, I have chosen the default to no interprocedural analysis for this DFA analysis for perf reasons. Users can turn it on through editorconfig options if they are making these platform checks in helpers.